serpy/serializer.py

```diff
@@ -112,7 +112,13 @@ def _serialize(self, instance, fields):
                         result = result()
                     if to_value:
                         result = to_value(result)
-            v[name] = result
+
+            if required:
+                v[name] = result
+            elif not required and result:
```

Comment:
so there's a problem here if the result is a falsy value, like 0 or '' it won't be in the serialized dict. I think this would be more correct as `result is not None`

You can combine the checks as:
```py
if required or result is not None:
    v[name] = result
```
The `else: continue` is also unnecessary since it's at the end of the loop anyway.

It would be great if you could add a test for the case where the field is not required but produces a falsy value like 0.